generatorreceiver/internal/topology/kubernetes.go

```diff
@@ -72,11 +74,30 @@ func (k *Kubernetes) CreatePod(serviceName string) {
 	k.Service = serviceName
 }
 
-func (k *Kubernetes) RestartPod() {
+func (k *Kubernetes) RestartIfNeeded(flags flags.EmbeddedFlags, logger *zap.Logger) {
+	if k == nil || k.Restart.Every == 0 {
+		return
+	}
+
 	k.mutex.Lock()
 	defer k.mutex.Unlock()
+
+	flagTime := flags.UpdatedTime()
```

Comment:
If `FlagSet` and `FlagUnset` are _both_ specified, `UpdatedTime()` will only return `FlagSet`'s updated time. Should we add a validation step that bans `FlagSet` and `FlagUnset` from both being specified? We would lose some flexibility - for example, we might want a pod to only run if `flag_a` is enabled _and_ `flag_b` is disabled. But this might complicate pod restart logic too much. @njvrzm do you have any opinions on this?

Reply:
I've changed the logic to return the update time from flag that was updated later. The reason for this is that flags that have both `set` and `unset` will only generate data when both are set to the correct state (`flag_set` should be active and `flag_unset` should be disabled) and added some tests for that.

Do you think that is enough to work in this case or am I missing something?

Reply:
I'm going to write out an example (to verify my understanding):
Let's say a pod has `flag_set=flag_a` and `flag_unset=flag_b`. When `GenerateStartTime()` is called, it will return 0 if the pod is not "running" (i.e. when`ShouldGenerate()` returns false). If the pod _is_ running (i.e. when `ShouldGenerate()` returns true), then it will return the time of whichever flag was updated later. If `flag_a` was more recently updated, that means the pod's `flag_set` went from disabled to enabled, and the pod went from not-running to running. If `flag_b` was more recently updated, that means the pod's `flag_unset` went from enabled to disabled, and the pod went from not-running to running.

So, yes I think that will solve it! 👍

Reply:
Hey, sorry I missed this yesterday! I don't love the way `FlagSet` and `FlagUnset` work, but I do think it's worth keeping the flexibility. And @carolynblumberg I think your reasoning is exactly right, I went through the same thought process 😂